File: zapp-common/src/main/java/com/zfoo/app/zapp/common/protocol/group/model/ChannelVO.java
/*
 * Copyright (C) 2020 The zfoo Authors
 *
 * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
 * in compliance with the License. You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed
 * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and limitations under the License.
 */

package com.zfoo.app.zapp.common.protocol.group.model;

import com.zfoo.protocol.IPacket;

import java.util.List;

/**
 * @author jaysunxiao
 * @version 1.0
 * @since 2020-04-22 11:43
 */
public class ChannelVO implements IPacket {

    public static final transient short PROTOCOL_ID = 18010;

    private long id;
    private String name;
    private long refreshTime;
    private List<ChannelAuthVO> channelAuths;

    public static ChannelVO valueOf(long id, String name, long refreshTime, List<ChannelAuthVO> channelAuths) {
        var vo = new ChannelVO();
        vo.id = id;
        vo.name = name;
        vo.refreshTime = refreshTime;
        vo.channelAuths = channelAuths;
        return vo;
    }

    @Override
    public short protocolId() {
        return PROTOCOL_ID;
    }


    public long getId() {
        return id;
    }

    public void setId(long id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public long getRefreshTime() {
        return refreshTime;
    }

    public void setRefreshTime(long refreshTime) {
        this.refreshTime = refreshTime;
    }

    public List<ChannelAuthVO> getChannelAuths() {
        return channelAuths;
    }

    public void setChannelAuths(List<ChannelAuthVO> channelAuths) {
        this.channelAuths = channelAuths;
    }
}

File: zapp-gateway/jsProtocol/group/model/ChannelVO.js
// @author jaysunxiao
// @version 1.0
// @since 2020-04-22 11:43
const ChannelVO = function(channelAuths, id, name, refreshTime) {
    this.channelAuths = channelAuths; // java.util.List<com.zfoo.app.zapp.common.protocol.group.model.ChannelAuthVO>
    this.id = id; // long
    this.name = name; // java.lang.String
    this.refreshTime = refreshTime; // long
};

ChannelVO.prototype.protocolId = function() {
    return 18010;
};

ChannelVO.write = function(buffer, packet) {
    if (buffer.writePacketFlag(packet)) {
        return;
    }
    buffer.writePacketArray(packet.channelAuths, 18012);
    buffer.writeLong(packet.id);
    buffer.writeString(packet.name);
    buffer.writeLong(packet.refreshTime);
};

ChannelVO.read = function(buffer) {
    if (!buffer.readBoolean()) {
        return null;
    }
    const packet = new ChannelVO();
    const list0 = buffer.readPacketArray(18012);
    packet.channelAuths = list0;
    const result1 = buffer.readLong();
    packet.id = result1;
    const result2 = buffer.readString();
    packet.name = result2;
    const result3 = buffer.readLong();
    packet.refreshTime = result3;
    return packet;
};

export default ChannelVO;
